kernel/dev/serial/rs232.cpp

```diff
@@ -15,10 +15,13 @@
 #include <arch/arch.hpp>
 #include <dev/serial/rs232.hpp>
 #include <mem/heap.hpp>
+#include <lib/mutex.hpp>
 
 static px_ring_buff_t* read_buffer = NULL;
 uint8_t rs_232_line_index;
 uint16_t rs_232_port_base;
+px_mutex_t mutex_data;
+px_mutex_t mutex_init;
```

Comment:
I'm concerned that we're not initializing these properly. I think the default of `0` will work for now, but we might want to change that in the future which would break this. Maybe we could create a macro that would declare and initialize these structures properly?

```c++
// in mutex.hpp
#define PX_MUTEX(name) px_mutex_t name = {}

// in .cpp files where we use mutexes
PX_MUTEX(mutex_data);
```

I'm open to other ideas as well.

Reply:
Yeah, we could do something similar to PTHREAD_MUTEX_INITIALIZER

Reply:
Why use a macro when you define this in a constructor?

A macro seems like both overkill and a way of subverting a final solution if we are just trying to abstract away initialization.

Reply:
Very true. We could actually use C++ constructors to make initialization completely transparent.

Reply:
Yeah, let's not fall into the same trap of the Linux kernel by depending on C-era macros that are mimicking C++ features.

Reply:
Very true. I keep forgetting that Panix is our opportunity to break the mold. We can always add in compatibility later if desired, but for now we can do things differently / better.

Reply:
Default constructors have been added in 533581f
Pinging @JamesOrson & @micahswitzer for their input.